### ISSUE TITLE
Different optimization criteria for matrix factorization in the code and in the paper

### ISSUE THREAD

**Issue by VConchello** [yzhao062/MetaOD#5]
In the paper, it’s said that MetaOD minimizes the sum of sDCG as the optimization criteria to factorise a matrix into latent factors (Section 3.4.1), but the code (_core.py:91,156,166_) is using the function `ndcg_score`, from `sklearn.metrics`, which is different than sDCG in some aspects.
And then for the gradient descent it uses the gradient of sDCG to find an optimum.
Is there any rationale for these changes?

**Comment by yzhao062** [comment]
The reason is more for numerical stability. You can replace that by dcg and the results should be almost the same (some numerical stability may mess up though). 
![image](https://user-images.githubusercontent.com/15079146/177804148-7fcfabe9-2c4f-46f9-ad4a-d4a46d8f465a.png)

So IDCG is more like a scaling factor but will not change the results of the ranking to my understanding.

**Comment by VConchello** [comment]
Thank you for your answer. The scaling part is clear, I think, but looking at the formula: 
![image](https://user-images.githubusercontent.com/108863240/177806204-ba73eb19-1363-4a7b-a813-391cce0f52a6.png)
First, this formula is using the sigmoid. And also the numerator is pretty different.
I'm not sure if these changes have an important impact on the evaluation of this metric.

**Comment by yzhao062** [comment]
That is because DCG is not differentiable...specifically log2 (i+1) -> i is an indicator function and has no derivatives. We use sigmoid to approximate it.

![image](https://user-images.githubusercontent.com/15079146/177810336-423bbb9e-83f1-4707-b854-07a1ab1ae122.png)

**Comment by VConchello** [comment]
Okay, I understood that the approximation would be used for both the gradient and the function to be minimised, not just to find the gradient.
Thank you for the answer.
